projects/project-2/assignment/reviewers/GDC_lv6.sparql

```diff
@@ -0,0 +1,24 @@
+# checking whether all GDCs are generically depending on something
+#There is probably a conceptual mistake here, since this should be axioms and not direct connections between classes
+# lv 6
+
+
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX obo: <http://purl.obolibrary.org/obo/>
+
+SELECT ?GDC (CONCAT("Warning: Generically Dependent Entity ", STR(?label), " doesn't generically depend on some entity.") AS ?warningMessage)
+WHERE {
+    #get all the subclasses of GDC
+    ?GDC rdfs:subClassOf+ obo:BFO_0000031 .
+    ?GDC rdfs:label ?label .
+    FILTER (!ISBLANK(?GDC)) .
+    
+    # check whether it doesn't generically depend on something
+    FILTER NOT EXISTS {
+        ?GDC rdfs:subClassOf [
+            rdf:type owl:Restriction ;
+            owl:onProperty obo:BFO_0000084 ;
+            owl:someValuesFrom ?targetClass 
+        ].
+}
```

Comment:
If I run the query of protégé I have an error (see attached)
<img width="695" alt="Screenshot 2024-10-10 alle 14 21 41" src="https://github.com/user-attachments/assets/83b02aab-41cd-4507-bf7e-494138f08c05">